### DISCUSSION TITLE
Integration with CalDAV server (like Nextcloud Calendar)

### DISCUSSION THREAD

**Discussion by delucca** [docker-mailserver/docker-mailserver#2402] [Q&A]
# Subject

How to integrate `docker-mailserver` with a calendar server

## Description

I would like to know how could I integrate `docker-mailserver` with a calendar server. I have both `dms` and `Nextcloud` running in my personal server and I would like to integrate them both.

My calendar is already working on `Nextcloud` calendar and I can send invitations to events accordingly (since I've configured `Nextcloud` to use my e-mail from `dms` as default). My problem is to **receive** invitations to events. I receive them in my e-mail (managed by `dms`) but when I accept those invitations I would like to automatically create the event in my CalDAV server (like what happens in Google Calendar, or most e-mail servers that have an embedded calendar)

Is that possible?

I found a few alternatives (in Mailcow and Mailinabox) but they seem to have their own Nextcloud/SOGo installation inside. It seems like a **huge** overkill for me. I think with some configs that should be doable.

Thanks!

**Comment by jayknyn** [comment]
Hi Daniel, I struggled a lot with this also, primarily trying to get a calendar integration in a webmail setup (I'm using RoundCube for webmail and I also have NextCloud). 

My working solution in the end was to use a desktop email client (I'm on mac, so macos mail app) and desktop calendar client with the calendar server being the CaldDAV server on NextCloud. So now when I get an invite in my email, I double-click the `.ics` file and it gets added to the desktop calendar client (choose the NextCloud CalDAV calendar) and then this event syncs to the NextCloud CalDAV server. I then have this synced to my mobile phone (Android so using Etar and DAVx5). 

All works pretty seemlessly now but yeah, it's not as smooth as using a groupware that bundles all this together. I've gotten my partner to use this setup and seems to be working well. Not sure I can convince less technically-minded users to use this setup though.

**Comment by delucca** [comment]
> Hi Daniel, I struggled a lot with this also, primarily trying to get a calendar integration in a webmail setup (I'm using RoundCube for webmail and I also have NextCloud).
> 
> My working solution in the end was to use a desktop email client (I'm on mac, so macos mail app) and desktop calendar client with the calendar server being the CaldDAV server on NextCloud. So now when I get an invite in my email, I double-click the `.ics` file and it gets added to the desktop calendar client (choose the NextCloud CalDAV calendar) and then this event syncs to the NextCloud CalDAV server. I then have this synced to my mobile phone (Android so using Etar and DAVx5).
> 
> All works pretty seemlessly now but yeah, it's not as smooth as using a groupware that bundles all this together. I've gotten my partner to use this setup and seems to be working well. Not sure I can convince less technically-minded users to use this setup though.

Thanks for your reply. That is a nice workaround :)

In any case, I would like to know if it is possible to do a more "clean" integration. It seems pretty complicated. I saw how Mailcow and MIAB handle their calendars, and they launch an entire Nextcloud instance just for the calendar O.O

**Comment by wernerfred** [comment]
Do you mind if i (or you) convert this Issue into a discussion? Imho this is more a general purpose question if someone of the community has archived smth similar rather than an (code related) issue with `dms`

**Comment by delucca** [comment]
> Do you mind if i (or you) convert this Issue into a discussion? Imho this is more a general purpose question if someone of the community has archived smth similar rather than an (programatic) issue with `dms`

Sure!

Do you know if it is possible to convert an issue to a discussion? I've never done that.
